nir/src/main/scala/scala/scalanative/nir/Types.scala

```diff
@@ -38,7 +38,7 @@ object Type {
   final object F64 extends F(64)
 
   final case class Array(ty: Type, n: Int)              extends Type
-  final case class Function(args: Seq[Type], ret: Type) extends Type
+  final case class Function(args: Seq[Arg], ret: Type)  extends Type
```

Comment:
Do you think calling convention should be added as part of this pull request, or should it be added later on?